error.go

```diff
@@ -20,295 +20,233 @@
 package goa
 
 import (
-	"bytes"
-	"encoding/json"
 	"fmt"
-	"strconv"
 	"strings"
 )
 
-type (
-	// ErrorID is an enum listing the possible types of errors.
-	ErrorID int
-
-	// TypedError describes an error that can be returned in a HTTP response.
-	TypedError struct {
-		ID   ErrorID
-		Mesg string
-	}
-
-	// MultiError records multiple errors.
-	MultiError []error
-
-	// BadRequestError is the type of errors that result in a response with status code 400.
-	BadRequestError struct {
-		Actual error
-	}
+const (
+	errInvalidParamTypeID     = "goa.ipt"
+	errMissingParamID         = "goa.mpa"
+	errInvalidAttributeTypeID = "goa.iat"
+	errMissingAttributeID     = "goa.mat"
+	errInvalidEnumValueID     = "goa.iev"
+	errMissingHeaderID        = "goa.mhe"
+	errInvalidFormatID        = "goa.ifo"
+	errInvalidPatternID       = "goa.ipa"
+	errInvalidRangeID         = "goa.ira"
+	errInvalidLengthID        = "goa.ile"
+	errInvalidEncodingID      = "goa.ien"
+	errInternalID             = "goa.int"
```

Comment:
two things here:

1. what is the use of such constants, since they are only used below, and only once ?
2. I'd be more comfortable having generic error codes, rather than codes showing the internal system I'm using. Also, `ipt` as a human-readable machine error code is not descriptive at all. I understand it comes with another message though, but what is its use ?